core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -679,6 +680,10 @@ class SnappyParser(session: SnappySession)
     }
   }
 
+  protected final def fetchExpression: Rule1[Expression] = rule {
+    FETCH ~ FIRST ~ TOKENIZE_END ~ expression ~ ((ROW|ROWS) ~ ONLY) ~> ((f: Expression) => f)
```

Comment:
Just curious, can we replace 'expression' with more specific i.e. numericLiteral? What error do we get now if limit/fetch-first value we get now is -ve or  double?

Reply:
Since the change is more like providing an alias to LIMIT clause so I used the same rule of expression which LIMIT is using.

Reply:
@vivekwiz LIMIT in Spark allows for arbitrary expressions that evaluate to a constant integer.